cmd/cosign/cli/attest/attest.go

```diff
@@ -116,15 +117,20 @@ func AttestCmd(ctx context.Context, ko options.KeyOpts, regOpts options.Registry
 	wrapped := dsse.WrapSigner(sv, types.IntotoPayloadType)
 	dd := cremote.NewDupeDetector(sv)
 
+	var predicate []byte
```

Comment:
What do you think about making this `var predicate io.ReadCloser` so you don't have to add the `bytes.NewReader` shim below?

Reply:
You'd then just set `predicate = os.Stdin` and `predicate, err = os.Open(predicatePath)`. Remember to `defer predicate.Close()` too. Feels a little cleaner, and doesn't use up all your memory if `predicatePath` contains a huge file (at least not until deeper in the call stack).